docs: add docs on general Terraform usage (#1263)
edgelesssys/constellation: docs/docs/reference/terraform.md

> # Terraform Usage

Reviewer: > the new page does not show up in the sidebar

You need to add this page to sidebars.js. If you then set the version to Next, it will show up.

[Terraform](https://www.terraform.io/) is an open-source Infrastructure as Code (IaC) framework which is being used by multiple Constellation components to manage cloud resources. This page describes our policy on using Terraform in Constellation.

:::info

> This page assumes basic familiarity with Terraform. The Terraform documentation can be found [here](https://developer.hashicorp.com/terraform/docs).

Reviewer 2: ```suggestion
This page assumes basic familiarity with Terraform. Refer to the [Terraform documentation](https://developer.hashicorp.com/terraform/docs) for an introduction.
```

Embedding link into more natural language.
